pkg/apis/configs/dataresidency/defaults.go

```diff
@@ -48,18 +53,35 @@ func (d *Defaults) AllowedPersistenceRegions() []string {
 	return d.scoped().AllowedPersistenceRegions
 }
 
+// Global gets the Global setting in the default.
+func (d *Defaults) Global() bool {
+	return d.scoped().Global
+}
+
 // ComputeAllowedPersistenceRegions computes the final message storage policy in
 // topicConfig. Return true if the topicConfig is updated.
-func (d *Defaults) ComputeAllowedPersistenceRegions(topicConfig *pubsub.TopicConfig) bool {
+func (d *Defaults) ComputeAllowedPersistenceRegions(topicConfig *pubsub.TopicConfig, clusterRegionGetter utils.ClusterRegionGetter) (bool, error) {
+	if topicConfig.MessageStoragePolicy.AllowedPersistenceRegions != nil {
+		// Don't try to change anything if it is not empty
+		return false, fmt.Errorf("topic config is unexpectedly set, please check the logic again")
+	}
```

Comment:
Is this expected to happen if I

1. Set `allowedpersistenceregions`.
1. Create a Broker/Source.
1. Change my `allowedpersistenceregions`

?

Reply:
In this case, we don't update existing ones in 1.

Reply:
I'm not sure I understand. I know that if the user changes the `allowedpersistencergions`, we _don't_ move anything that already exists. I think this is what your comment means, is that correct?

My overall question is: what happens to the Broker/Source that is using the older location? Does it continue to reconcile correctly? Or does it start failing reconciliation due to this error? My expectation is that it would continue to reconcile correctly, using its original location, probably with a warning in its status saying that it is using an older location (maybe a K8s event too?).

Reply:
Your understanding about my comment is correct. Since we don't reconcile existing topics, the only way we invoke this is when we are creating topics, so this is to avoid changing anything here in case in the future we somehow allow other configuration before getting the default. Eg. if we have a data residency field in the broker spec. The errors in this function are treated as a warning in the callers. So there will never be failure because of this. The original reasoning is that resolving final AllowedPersistenceRegions is best effort, but I guess maybe we should fail in case of unable to get the cluster region. How about 1) checking the error and fail the topic creation, and 2) change this to no error?